pkg/kubelet/nodestatus/setters.go

```diff
@@ -160,16 +162,19 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			} else if addr := net.ParseIP(hostname); addr != nil {
 				ipAddr = addr
 			} else {
-				var addrs []net.IP
-				addrs, _ = net.LookupIP(node.Name)
+				ctx, cancel := context.WithTimeout(context.Background(), nodeStatusUpdateFrequency/2)
+				defer cancel()
+
+				addrs, _ := net.DefaultResolver.LookupIPAddr(ctx, node.Name)
```

Comment:
we should be careful not to change the current behavior, is `net.LookupIP` and `net.DefaultResolver.LookupIPAddr(` the same?

Reply:
if you dig into the implementation of `net.LookupIP`, you'll find the answer is yes.

Reply:
my apologies, I just found you already clarified here
https://github.com/kubernetes/kubernetes/pull/80155#discussion_r303710271